### PR TITLE
Add progress for deck flashing

### PR DESCRIPTION
**Pull request by jonasdn**
- bootloader: Remove spurious print
- Add progress callback to flashing decks

### CODE REVIEW

File: cflib/crazyflie/mem/__init__.py [hunk 1]
```diff
@@ -203,6 +207,12 @@ def write_done(self, addr):
                 'Address did not match when adding data to read request!')
             return
 
+        if self._progress_cb is not None:
+            new_progress = 100 * (self._write_len - self._bytes_left) / self._write_len
+            if int(new_progress) > self._progress:
+                self._progress = int(new_progress)
+                self._progress_cb(f'Writing to memory (id: {self.mem.id})',
```
Comment (ataffanel) [inline cflib/crazyflie/mem/__init__.py:214]: Is it possible to get a more descriptive name for the memory there (ie. the name of de deck for a deck memory)? Would be more user friendrly than the ID (This ID is an implementation detail)

**Review by ataffanel — COMMENTED** [review]
Looks like a great addition to the memory subsystem!

I just have one comment about the callback information, it should maybe containt a bit more user-friendly information.

File: cflib/crazyflie/mem/__init__.py (hunk 1 above)
Comment (jonasdn) [reply cflib/crazyflie/mem/__init__.py]: I added special casing of the progress message, to get a better message, such as:

```
$ python3 -m cfloader flash -w radio://0/80/2M/E7E7E7E7E8 ~/Downloads/nightly/aideck-NINA-firmware.bin deck-bcAI-fw
Reset to bootloader mode ...
Could not save cache, no writable directory
Could not save cache, no writable directory
◢ 0% Writing to bcAI deck memory
◣ 1% Writing to bcAI deck memory
◤ 2% Writing to bcAI deck memory
◥ 3% Writing to bcAI deck memory
◢ 4% Writing to bcAI deck memory
◣ 5% Writing to bcAI deck memory
◤ 6% Writing to bcAI deck memory
◥ 7% Writing to bcAI deck memory
◢ 8% Writing to bcAI deck memory
◣ 9% Writing to bcAI deck memory
◤ 10% Writing to bcAI deck memory
```

Comment (jonasdn) [reply cflib/crazyflie/mem/__init__.py]: cfclient-bcai

**Review by knmcguire — APPROVED** [review]

**Review by krichardsson — APPROVED** [review]

**Pull request merged**